src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/PaperResponseAdmissionCallbackHandlerTest.java

```diff
@@ -0,0 +1,366 @@
+            .thenReturn(docAssemblyResponse);
+        when(docAssemblyTemplateBodyMapper.paperResponseAdmissionLetter(any(CCDCase.class), any(String.class)))
+            .thenReturn(DocAssemblyTemplateBody.builder().build());
+        when(docAssemblyResponse.getRenditionOutputLocation()).thenReturn(DOC_URL);
+        when(documentManagementService.getDocumentMetaData(anyString(), anyString())).thenReturn(getLinks());
+        when(clock.instant()).thenReturn(LocalDate.parse("2020-06-22").atStartOfDay().toInstant(ZoneOffset.UTC));
+        when(clock.getZone()).thenReturn(ZoneOffset.UTC);
+        when(clock.withZone(LocalDateTimeFactory.UTC_ZONE)).thenReturn(clock);
+        when(caseDetailsConverter.extractCCDCase(any(CaseDetails.class))).thenReturn(ccdCase);
+        ArgumentCaptor<CCDCase> ccdDataArgumentCaptor = ArgumentCaptor.forClass(CCDCase.class);
+        handler.handle(callbackParams);
+        verify(caseDetailsConverter).convertToMap(ccdDataArgumentCaptor.capture());
+
+        verify(documentManagementService, times(1))
+            .getDocumentMetaData(anyString(), anyString());
+
+        verify(docAssemblyTemplateBodyMapper, times(1)).paperResponseAdmissionLetter(any(CCDCase.class),
+            any(String.class));
+
+        assertEquals("CMC-defendant-case-handoff.pdf", ccdDataArgumentCaptor.getValue().getCaseDocuments().get(0)
+            .getValue().getDocumentLink().getDocumentFileName());
+        assertEquals(GENERAL_LETTER,
+            ccdDataArgumentCaptor.getValue().getCaseDocuments().get(0).getValue().getDocumentType());
+        assertEquals(1, ccdDataArgumentCaptor.getValue().getCaseDocuments().size());
+    }
+
+    @Test
+    void shouldThrowExceptionWhenGenerateAndUpdateCaseDocumentFails() {
+
+        CCDCase ccdCase = getCCDCase(PART_ADMISSION, CCDRespondent.builder(), "OCON9x");
+
+        Claim claim = Claim.builder()
+            .referenceNumber("XXXXX")
+            .build();
+        when(caseMapper.from(any(CCDCase.class))).thenReturn(claim);
+        when(userService.getUserDetails(AUTHORISATION)).thenReturn(userDetails);
+        when(docAssemblyService
+            .renderTemplate(any(CCDCase.class), anyString(), anyString(), any(DocAssemblyTemplateBody.class)))
+            .thenThrow(new RuntimeException("exception"));
+        when(docAssemblyTemplateBodyMapper.paperResponseAdmissionLetter(any(CCDCase.class), any(String.class)))
+            .thenReturn(DocAssemblyTemplateBody.builder().build());
+        when(caseDetailsConverter.extractCCDCase(any(CaseDetails.class))).thenReturn(ccdCase);
+
+        Assertions.assertThrows(RuntimeException.class,
+            () -> handler.handle(callbackParams));
+    }
+
+    @Test
+    public void shouldHaveCorrectCaseworkerRole() {
+        assertThat(handler.getSupportedRoles()).containsOnly(CASEWORKER);
+    }
+
+    @Test
+    public void shouldHaveCorrectCaseworkerEvent() {
```

Comment:
Access should be default for Junit 5 tests
```suggestion
   void shouldHaveCorrectCaseworkerEvent() {
```